### ISSUE TITLE
Fix potential dependency mismatch with Spring Boot 3 and Applications Insights for Spring native

### ISSUE THREAD

**Issue by jeanbisutti** [Azure/azure-sdk-for-java#35727]
Dependency management of _Application Insights for Spring native_ declares OTel BOMs.

Spring Boot 3 also use OTel dependencies. 

In a project with Spring Boot 3 and Application Insights for Spring native, some of the versions of the OTel dependencies come from Spring Boot 3. To make _Application Insights for Spring native_ works with native images and all Spring Boot 3 projects, the following properties are added in  _Application Insights for Spring native_ :
```
--initialize-at-build-time=io.opentelemetry.instrumentation.logback.appender.v1_0.internal.LoggingEventMapper \
 --initialize-at-build-time=io.opentelemetry.instrumentation.api.internal.cache.concurrentlinkedhashmap.ConcurrentLinkedHashMap
```
These properties are supposed to come from the OTel Java project.

Project having the decribed issue:  https://github.com/jdubois/NubesGen-2/tree/app-insights-native

Related issue: https://github.com/open-telemetry/opentelemetry-java/issues/5338

**Comment by jeanbisutti** [comment]
From Spring Boot 3, the `spring-boot-dependencies` BOM referecences the `opentelemetry-bom` BOM.  The OTel vesion is different in `Spring Monitor`.

The align the version of the OTel dependencies, the user could 

1) Set the `opentelemetry.version` Maven property declared in the SPring Boot Boom

```
<properties>
   <opentelemetry.version>1.28.0</opentelemetry.version>
</properties>
```

OR

2) Declare `opentelemetry-bom`

```XML
 <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.opentelemetry</groupId>
        <artifactId>opentelemetry-bom</artifactId>
        <version>1.28.0</version>
        <type>pom</type>
      </dependency>
    </dependencies>
  </dependencyManagement>
```

**Comment by jeanbisutti** [comment]
Fixed with #36610
